ironfish/src/network/peerNetwork.ts

```diff
@@ -396,12 +399,14 @@ export class PeerNetwork {
    * Call close when shutting down the PeerNetwork to clean up
    * outstanding connections.
    */
-  stop(): void {
-    this.started = false
-    this.peerConnectionManager.stop()
-    this.peerManager.stop()
-    this.webSocketServer?.close()
-    this.updateIsReady()
+  async stop(): Promise<void> {
+    await Promise.allSettled([
+      (this.started = false),
+      this.peerConnectionManager.stop(),
+      this.peerManager.stop(),
+      this.webSocketServer?.close(),
+      this.updateIsReady(),
+    ])
```

Comment:
This is creating a bug now, if any of these fail to stop the stop() will resolve successfully.

Also why is the non async thing inside of this promise?

Reply:
That was probably a holdover from a prior implementation that I had attempted; I'll remove.